### PR TITLE
Forgot password option

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-158
| Versions      | 3.3

### CODE REVIEW

File: docs/guide/user_management/user_management.md [hunk 1]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
 The validity of the password recovery token can be set using the `ezplatform.system.<siteaccess>.security.token_interval_spec` parameter.
 By default it is set to `PT1H` (one hour).
 
+### Add forget password option in site front
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:20]:
```suggestion
### Add forget password option
```

File: docs/guide/user_management/user_management.md [hunk 2]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
...
+
+You can add a forgot password option to the site front to allow users of a specific site request a password change.
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:22]:
```suggestion
You can add a "forgot password" option to the site front to allow users of a specific site request a password change.
```

File: docs/guide/user_management/user_management.md [hunk 3]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
...
+
+Edit `congfig/packages/ezplatform.yaml`. The templates are defined with the following configuration:
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:24]:
```suggestion
Edit `config/packages/ezplatform.yaml`. The templates are defined with the following configuration:
```

File: docs/guide/user_management/user_management.md [hunk 4]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
...
+
+```yaml
+ezplatform:
+    system:
+        <siteaccess>:
+            user_forgot_password:
+                templates:
+                    form: <path_to_template>
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:32]: Maybe one sentence explaining what the two templates are responsible for? Obvious, I know, but...

File: docs/guide/user_management/user_management.md [hunk 5]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
...
+                    mail: <path_to_template>
+```
+
+The [templates](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/views/themes/admin/account/forgot_password/) for forgot password form and email are located in `/src/bundle/Resources/views/themes/admin/account/` in `ezsystems/ezplatform_admin_ui`.
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:36]:
```suggestion
The [default templates](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/views/themes/admin/account/forgot_password/) for forgot password form and email are located in `/src/bundle/Resources/views/themes/admin/account/` in `ezsystems/ezplatform_admin_ui`.
```
... I think - are these the general default templates, or only Back-Office-related? I see we also have other templates for this here: https://github.com/ezsystems/ezplatform-user/blob/master/src/bundle/Resources/views/forgot_password/index.html.twig

File: docs/guide/user_management/user_management.md [hunk 6]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
...
+
+**Forgot password forms:**
+
+```yaml
+user_change_password.templates.success
+user_forgot_password.templates.form
+user_forgot_password.templates.mail
+user_forgot_password_success.templates.form
+user_forgot_password_login.templates.form
+user_reset_password.templates.form
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:48]: These are actually mentioned already in https://doc.ibexa.co/en/master/guide/user_management/user_management/#other-user-management-templates

**Review by DominikaK — CHANGES_REQUESTED** [review]
Also need info on how to add a link leading to the reset password form somewhere in other templates. (Maybe it's just a link to the route?)

File: docs/guide/user_management/user_management.md (hunk 5 above)
Comment (juskora) [reply docs/guide/user_management/user_management.md:36]: These are fallback templates.

File: docs/guide/user_management/user_management.md (hunk 6 above)
Comment (ViniTou) [reply docs/guide/user_management/user_management.md:48]: They also gonna have proper semantic configuration once https://github.com/ezsystems/ezplatform-user/pull/88 is merged. On the other hand in this state those params names are not complete here. I would rather put full semantic configuration somewhere.

File: docs/guide/user_management/user_management.md [hunk 7]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
...
+ezplatform:
+    system:
+        <siteaccess>:
+            user_forgot_password:
+                templates:
+                    form: <path_to_template>
+                    mail: <path_to_template>
+```
+
+The [default templates](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/views/themes/admin/account/forgot_password/) for forgot password form and email are located in `/src/bundle/Resources/views/themes/admin/account/` in `ezsystems/ezplatform_admin_ui`.
```
Comment (ViniTou) [inline docs/guide/user_management/user_management.md:36]: To be precise the defaults one are located in https://github.com/ezsystems/ezplatform-user/tree/master/src/bundle/Resources/views, those in adminUI are specific one to our adminUI (I would mention both in docs).

File: docs/guide/user_management/user_management.md [hunk 8]
```diff
@@ -17,6 +17,41 @@ You can [customize it according to your needs](#customize-login-form).
 The validity of the password recovery token can be set using the `ezplatform.system.<siteaccess>.security.token_interval_spec` parameter.
 By default it is set to `PT1H` (one hour).
 
+### Add forget password option
+
+You can add a "forgot password" option to the site front to allow users of a specific site request a password change.
```
Comment (ViniTou) [inline docs/guide/user_management/user_management.md:22]: I would rather say something along lines, that you can customize template used for `/user/forgot-password` for specific siteaccess, those can be either front or admin ones.

File: docs/guide/user_management/user_management.md [hunk 9]
```diff
@@ -17,6 +17,35 @@ You can [customize it according to your needs](#customize-login-form).
...
+```
+
+Under the `templates` key, provide the path to templates responsible for rendering the forgot password form (`form`) and email (`mail`), which is sent to users after they request a password change.
+
+The [default templates](https://github.com/ezsystems/ezplatform-user/tree/master/src/bundle/Resources/views) for forgot password form and email are located in `ezplatform-user/src/bundle/Resources/views`.
+The [templates](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Resources/views/themes/admin/account/forgot_password/) specific for the Back Office are in `ezplatform-admin-ui/src/bundle/Resources/views/themes/admin/account`.
+
+You can also modify [other user management templates](#other-user-management-templates).
+
+To add link redirecting to the reset password form, in the landing page template, provide the following code:
```
Comment (DominikaK) [inline docs/guide/user_management/user_management.md:43]:
```suggestion
To add link redirecting to the reset password form, in the template, provide the following code:
```
The link can work from any other template, not just landing page

**Review by DominikaK — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Pull request merged**